runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -1957,10 +2160,24 @@ J9::TransformUtil::transformIndirectLoadChainImpl(TR::Compilation *comp, TR::Nod
             }
          else if (symRef->getSymbol()->isCollectedReference())
             {
-            uintptr_t value = fej9->getReferenceFieldAtAddress((uintptr_t)fieldAddress);
-            if (value)
+            TR::KnownObjectTable::Index knotIndex = TR::KnownObjectTable::UNKNOWN;
+#if defined(J9VM_OPT_JITSERVER)
+            if (isServer)
+               {
+               knotIndex = ((J9::TransformUtil::value *)valuePtr)->idx;
+               }
+            else
+#endif /* defined(J9VM_OPT_JITSERVER) */
+               {
+               uintptr_t value = fej9->getReferenceFieldAtAddress((uintptr_t)valuePtr);
+               knotIndex = comp->getKnownObjectTable()->getOrCreateIndexAt(&value,
```

Comment:
`getOrCreateIndexAt(&value, isArrayWithConstantElements(symRef, comp));` decomposes into two operations:
```
   (1)  Index result = self()->getOrCreateIndexAt(objectReferenceLocation);
   (2) if (isArrayWithConstantElements) self()->addArrayWithConstantElements(result);
```
For the server mode we need to ensure that the second operation is done at the server, because when reading such information the server can do it without a message. So, after receiving the info from the client after doing
`comp->getKnownObjectTable()->updateKnownObjectTableAtServer(value, objectReferenceLocationClient);`
the server must also update the _arrayWithConstantElements in its own knot.